packages/mui-utils/src/resolveProps.ts

```diff
@@ -11,10 +20,14 @@ export default function resolveProps<
     slots?: Record<string, unknown>;
     slotProps?: Record<string, unknown>;
   } & Record<string, unknown>,
->(defaultProps: T, props: T) {
+>(defaultProps: T, props: T, mergePropsWhiteList?: Array<string>) {
   const output = { ...props };
 
   (Object.keys(defaultProps) as Array<keyof T>).forEach((propName) => {
+    const isMergeable = !!mergePropsWhiteList?.includes(String(propName));
```

Comment:
Can you move this to `if` statement below? I think it is more readable since `isMergeable` is not used anywhere.
```suggestion
```